core/templates/services/cookie.service.ts

```diff
@@ -0,0 +1,49 @@
+// Copyright 2021 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Service for manipulating getting and putting the cookies.
+ */
+
+/**
+ * NOTE: In general, we don't prefix our services with the 'oppia' prefix. This
+ * is an exception as the class name without oppia would collide with the
+ * existing class of CookieService from 'ngx-cookie'.
+ */
+
+import { Injectable } from '@angular/core';
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { CookieOptions, CookieService } from 'ngx-cookie';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class OppiaCookieService {
+  constructor(private ngxCookieService: CookieService) { }
+
+  getCookie(key: string): string | null {
+    if (!this.ngxCookieService.hasKey(key)) {
+      return null;
+    }
+    return this.ngxCookieService.get(key);
+  }
```

Comment:
Do we really need this service?
- `putCookie` function is redundant here.
- `getCookie` only handles the case where the cookie is not available. 

`ngxCookieService.get(key)` will return `undefined` for this case which makes more sense because the cookie is not available whereas `null` will indicate that cookie is available but no value is set on it yet.

I recommend removing this service, and using the service provided by `ngx-cookie`.
The service provided by `ngx-cookie` also has more functions e.g. hasKey, getObject, getAll, putObject, remove, removeAll. There can be a case in future, where a dev wants to use these functions.

He will have two ways to handle this.
- Use the `ngxCookieService` directly, this approach will introduce using two services for handling cookies which will be confusing and will introduce a bad structure in the codebase.
e.g. You went into this causing confusion issue in this PR itself here.
![image](https://user-images.githubusercontent.com/24855641/125288610-f6e96a80-e33b-11eb-8070-5f9242c6b1a0.png)
- Add a new function to OppieCookieService which calls the required function of `ngxCookieService`. This will lead to redundancy. You went this way when you wanted to use `putCookie`. You could have used `ngxCookieService` directly but you didn't want to create confusion.

If this is about not allowing undefined for typescript strict mode then we should leave it to the dev to handle this case. I don't think there will be a case where the dev will need to return `ngxCookieService.get(key)` from a function. Mostly, the dev will get the cookie value and perform operations on it in the function itself.
e.g.
![image](https://user-images.githubusercontent.com/24855641/125289700-29e02e00-e33d-11eb-8ef7-bd196fb444f1.png)

Reply:
Hi @ashutoshc8101, thanks for leaving a review! I think it is reasonable to remove this service and use the third-party library directly where required. Thanks for fixing this in https://github.com/oppia/oppia/pull/13403!